Sources/FoundationEssentials/FileManager/FileOperations.swift

```diff
@@ -903,6 +911,43 @@ enum _FileOperations {
     }
     #endif
 
+    #if os(WASI)
+    private static func _linkOrCopyFile(_ srcPtr: UnsafePointer<CChar>, _ dstPtr: UnsafePointer<CChar>, with fileManager: FileManager, delegate: some LinkOrCopyDelegate) throws {
+        var stat = stat()
+        guard lstat(srcPtr, &stat) == 0, !stat.isDirectory else {
```

Comment:
We might want to consult the delegate here - calling `shouldPerformOnItemAtPath` first, and if failing try `throwIfNecessary` and just return otherwise if the delegate wants to continue past this directory

Reply:
The first piece looks good, but I think we still need to check with the delegate before throwing below via `delegate.throwIfNecessary` - that way if the delegate wants to continue, we correctly continue on past this item

Reply:
Oops, I missed the second part.